lib/net.js

```diff
@@ -667,6 +640,16 @@ function onread(nread, buffer) {
     maybeDestroy(self);
   }
 
+  self.once('end', function end() {
+    maybeDestroy(this);
+  });
+  self.read(0);
+
+  if (!self.allowHalfOpen) {
+    self.write = writeAfterFIN;
+    self.destroySoon();
+  }
+
   // internal end event so that we know that the actual socket
   // is no longer readable, and we can start the shutdown
   // procedure. No need to wait for all the data to be consumed.
```

Comment:
Can we remove this event (`'_socketEnd'`) or is it kept to avoid possible breakage?

Reply:
@lpinca It’s still used in another location … I think after https://github.com/nodejs/node/pull/18608 things could be refactored that way, yes

Reply:
Gotcha.